File: leetCode/java/PalindromeNumber.java
// Given an integer x, return true if x is a palindrome, and false otherwise.

public class PalindromeNumber {

    public static boolean isPalindrome(int x) {

        if (x < 0) {
            return false;
        }

        var reverse = 0;
        var num = x;

        while (num > 0) {
            int lastDigit = num % 10;
            reverse = reverse * 10 + lastDigit;
            num = num / 10;
        }

        return reverse == x;
    }

    public static void main(String[] args) {
        System.out.println(isPalindrome(121));
        System.out.println(isPalindrome(-121));
        System.out.println(isPalindrome(10));
    }
}

File: leetCode/js/palindromeNumber.js
// Given an integer x, return true if x is a palindrome, and false otherwise.

var isPalindrome = function (x) {
  if (x < 0) {
    return false;
  }

  let reverse = 0;
  let input = x;

  while (input > 0) {
    const lastDigit = input % 10;
    reverse = reverse * 10 + lastDigit;
    input = Math.floor(input / 10);
  }

  return x === reverse;
};

console.log(isPalindrome(121));
console.log(isPalindrome(-121));
console.log(isPalindrome(10));
